scripts/india_aqi/preprocess.py

```diff
@@ -0,0 +1,304 @@
+            "Nov": "11",
+            "Dec": "12"
+        }
+        years = {
+            "2012": "12",
+            "2013": "13",
+            "2014": "14",
+            "2015": "15",
+            "2016": "16",
+            "2017": "17",
+            "2018": "18",
+            "2019": "19",
+            "2020": "20",
+            "2021": "21",
+            "2022": "22"
+        }
+        format_data = "%d/%m/%y %H:%M"
+        date_time = ""
+        dt = date.split("-")
+        date_time += dt[0]
+        date_time += "/"
+        date_time += months[dt[1]]
+        date_time += "/"
+        date_time += years[dt[2]]
+        date_time += " "
+        date_time += time
+        return datetime.strptime(date_time, format_data).isoformat()
+
+    def load(self):
+        if self.source.endswith("csv"):
+            df = pd.read_csv(self.source)
+
+        dcid_file = "India_AQI-stations_dcids_lgd.csv"
+        dcid_file_path = os.path.join(
+            os.path.dirname(__file__),
+            "data/{data_file}".format(data_file=dcid_file),
+        )
+        dcid = pd.read_csv(dcid_file_path)
+
+        df["dcid"] = ""
+        df["LgdCode"] = ""
+        df["date"] = ""
+        df["Year"] = ""
+
+        #changing the date format and renaming the columns
+        for i in range(df[df.columns[0]].count()):
+            df.iloc[[i], [32]] = df.iloc[i]["from_date"].split("-")[2]
+            df.iloc[[i], [31]] = self.ret_iso_format(df.iloc[i]["from_date"],
+                                                     df.iloc[i]["from_time"])
+        df.drop(['query_name', 'from_date', 'from_time', 'to_date', 'to_time'],
+                axis=1,
+                inplace=True)
+        df.rename(columns={
+            'pm25': 'PM25',
+            'pm10': 'PM10',
+            'no2': 'NO2',
+            'nh3': 'NH3',
+            'so2': 'SO2',
+            'co': 'CO',
+            'o3': 'O3',
+            'at': 'AT',
+            'bp': 'BP',
+            'sr': 'SR',
+            'rh': 'RH',
+            'wd': 'WD',
+            'no': 'NO',
+            'nox': 'NOx',
+            'benzene': 'Benzene',
+            'toluene': 'Toluene',
+            'xylene': 'Xylene',
+            'mp_xylene': 'MP_Xylene',
+            'eth_benzene': 'Eth_Benzene'
+        },
+                  inplace=True)
+
+        #adding dcid and LgdCode by mapping the stations with sites
+        length = dcid[dcid.columns[0]].count()
+        length2 = df[df.columns[0]].count()
+
+        for i in range(length):
+            for j in range(length2):
+                if (df.iloc[j]["site"] == dcid.iloc[i]["site_id"]):
+                    df.iloc[[j], [24]] = dcid.iloc[i]["dcid"]
+                    df.iloc[[j], [25]] = dcid.iloc[i]["LgdCode"]
+
+        #dropping id and site columns
+        df.drop(['id', 'site'], axis=1, inplace=True)
+
+        #rearranging columns
+        df = df[[
+            'Year', 'city', 'site_name', 'state', 'date', 'dcid', 'LgdCode',
```

Comment:
Yes sir, the column names are same, but the cases or names for 3-4 variables will actually change, for example from "city" to "City", and since the final rearrangement of columns come under processing part, i thought we can use the COLUMN_HEADERS in that part

Reply:
The key is to use constants which are defined somewhere clearly in the file. So if these are different compared to COLUMN_HEADERS, then let's define better constants for this. But hardcoding strings is not right.